Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -11,19 +11,61 @@ static void Main(string[] args)
             Test("42*?7=1974", 4);
             Test("42*?47=1974", -1);
             Test("2*12?=247", -1);
-            Console.ReadKey(true);
+            //Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(String equation){
+            int A,B,C;
+            String[] str=equation.Split("*");
+            String[] str1=str[1].Split("=");
+            if(str[0].Contains('?')){
+                B=Convert.ToInt32(str1[0]);
+                C=Convert.ToInt32(str1[1]);
+                if(B==0)
+                return -1;
+                if(C%B!=0)
+                return -1;
+                A=C/B;
+                return compareAndFind(A,str[0]);
+            }
+            else if(str1[0].Contains('?')){
+                A=Convert.ToInt32(str[0]);
+                C=Convert.ToInt32(str1[1]);
+                if(A==0)
+                return -1;
+                if(C%A!=0)
+                return -1;
+                B=C/A;
+                return compareAndFind(B,str1[0]);
+            }
+            else{
+                A=Convert.ToInt32(str[0]);
+                B=Convert.ToInt32(str1[0]);
+                C=A*B;
+                return compareAndFind(C,str1[1]);
+            }
+            throw new NotImplementedException();

```

Comment:
NotImplementedException is used for methods which aren't implemented. Here you have implemented method, so you should remove this.